pgmpy/factors/FactorSet.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+from pgmpy.factors import Factor
+from functools import reduce
```

Comment:
@ankurankan ```reduce``` is built-in in ```Python2.7``` not in ```Python3```. https://docs.python.org/3.0/whatsnew/3.0.html#builtins